pkg/asset/imagebased/image/ignition.go

```diff
@@ -0,0 +1,183 @@
+package image
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/coreos/ignition/v2/config/merge"
+	"github.com/coreos/ignition/v2/config/v3_2"
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+const (
+	ibiConfigurationPath     = "/var/tmp/ibi-configuration.json"
+	installationScriptPath   = "/usr/local/bin/install-rhcos-and-restore-seed.sh"
+	installationServiceName  = "install-rhcos-and-restore-seed.service"
+	networkConfigServiceName = "network-config.service"
+	pullSecretPath           = "/var/tmp/pull-secret.json" //nolint:gosec // not a secret, just a filename
+
+	trustedBundlePath        = "/etc/pki/ca-trust/source/anchors/additional-trust-bundle.pem"
+	nmstateConfigPath        = "/var/tmp/network-config.yaml"
+	registriesConfPath       = "/etc/containers/registries.conf"
+	postDeploymentScriptPath = "/var/tmp/post.sh"
+)
+
+// Ignition is an asset that generates the image-based installer ignition file.
+type Ignition struct {
+	Config *igntypes.Config
+}
+
+// Name returns the human-friendly name of the asset.
+func (i *Ignition) Name() string {
+	return "Image-based Installer Ignition"
+}
+
+// Dependencies returns the assets on which the Ignition asset depends.
+func (i *Ignition) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&ImageBasedInstallationConfig{},
+		&InstallationService{},
+		&RegistriesConf{},
+		&PostDeployment{},
+	}
+}
+
+type ibiConfigurationFile struct {
+	ExtraPartitionLabel  string `json:"extraPartitionLabel,omitempty"`
```

Comment:
Should I just add a default value for this field (i.e. [this](https://github.com/openshift-kni/lifecycle-agent/blob/main/api/ibiconfig/ibiconfig.go#L20) one) or remove it from the [config](https://github.com/openshift/installer/pull/8679/files#diff-9aed03fc6e829a8bce1cc7d68b6f2b2438d2bc852c025ae46aca4c58701f0d8dR25-R28) so that users cannot change its value and hardcode the default value?

Reply:
Same as https://github.com/openshift/installer/pull/8679#discussion_r1662899435.